.github/workflows/test-build-deploy.yml

```diff
@@ -0,0 +1,117 @@
+name: ci
+on:
+  push:
+    branches: [master]
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+** # Tag filters not as strict due to different regex system on Github Actions
+  pull_request:
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/cortexproject/build-image:update-golang-1.14.9-eb0c8d4d2
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v1
+      - name: Sym Link Expected Path to Workspace
+        run: |
+          mkdir -p /go/src/github.com/cortexproject/cortex
+          ln -s $GITHUB_WORKSPACE/* /go/src/github.com/cortexproject/cortex
+      - name: Generate protobuf files
+        run: "protoc -I /go/src:./vendor/github.com/thanos-io/thanos/pkg:./vendor/github.com/gogo/protobuf:./vendor:./pkg/ruler --gogoslick_out=plugins=grpc,Mgoogle/protobuf/any.proto=github.com/gogo/protobuf/types,:./pkg/ruler ./pkg/ruler/ruler.proto"
```

Comment:
Why is this required? The `make BUILD_IN_CONTAINER=false check-protos` should already build them. Have you had any issue?

My main concern here is that this is a copy of what we already have in the `Makefile` and we would have to keep it synched whenever we add protobuf definitions.

Reply:
tl;dr: our mistake mb! It's removed now


We couldn't see that line in the [lint command](https://github.com/cortexproject/cortex/blob/master/Makefile#L141) in the makefile so we added it manually. The line also didn't show up in console output under the Lint section
![Screen Shot 2020-10-09 at 3 12 40 PM](https://user-images.githubusercontent.com/20804975/95623825-e9692200-0a43-11eb-8d71-2dfa5a267742.png)

so we kept it in. But on a second look I realize I missed the `lint: protos` [line](https://github.com/cortexproject/cortex/blob/master/Makefile#L81)  and if we remove line 20, the `protoc -i` is run in `make lint` and the output shows in the console
![Screen Shot 2020-10-09 at 3 16 11 PM](https://user-images.githubusercontent.com/20804975/95623838-f259f380-0a43-11eb-84c9-4bfb79cf8812.png)